packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -31,7 +31,7 @@ class AnalyzeCommand extends FlutterCommand {
   @override
   Future<int> runInProject() async {
     Set<String> pubSpecDirectories = new Set<String>();
-    List<String> dartFiles = argResults.rest.toList();
+    Set<String> dartFiles = new Set<String>.from(argResults.rest);
```

Comment:
OK. It looks like the user can pass in a list of files, and then we add a bunch based on the flags and the directory structure. We may want to guard against duplicates at some point.